nxcomp/src/Timestamp.cpp

```diff
@@ -44,34 +44,55 @@
 
 T_timestamp timestamp;
 
-//
-// The following functions all use the ctime
-// static buffer from the C library.
-//
-
-char *strTimestamp(const T_timestamp &ts)
+std::string strTimestamp(const T_timestamp &ts)
 {
-  char *ctime_now = ctime((time_t *) &ts.tv_sec);
+  std::string ret;
+
+  char ctime_now[26] = { };
+  bool err = true;
+
+#if HAVE_CTIME_S
+  errno_t retval = ::ctime_s(ctime_now, 26, static_cast<const time_t*>(&ts.tv_sec));
+
+  err = (retval != 0);
+#else
+  char *retval = ::ctime_r(static_cast<const time_t*>(&ts.tv_sec), ctime_now);
 
+  err = (!(retval));
+#endif
+
+  if (err)
```

Comment:
drop err and move the if into the ifdef

Reply:
Why? That would duplicate the code for no good reason, since it would have to be in both the `#if` and `#else` branches.

Reply:
the err is basically the result of the active `#ifdef` block. If there was only one of the blocks, would you still use the err variable? Probably not. My suggestion will save two or three lines of code. (only the `if` line should be moved to the `#ifdef`, not the whole `if` block

Reply:
Okay, I finally understood what you mean. Yeah, I've been using `err` only to map the error code correctly since the two functions return different values (esp. in the error case).

Reply:
Done.